lib/cot/cottypes/positionreport.go

```
@@ -0,0 +1,95 @@
+package cottypes
+
+import (
+	"time"
+
+	"github.com/Novetta/go.datatypes/cot"
+	"github.com/Novetta/go.logging"
+	"github.com/Novetta/go.util"
+)
+
+/*
+	Defines Go Structs derived from schemas/PositionReport.xsd
+
+	types cross referenced with schemas/common.xsd
+*/
+
+// PositionReport defines a basic information and position report for non-hostile units.
+type PositionReport struct {
+	Unit       *Unit         `xml:"Unit,omitempty"`
+	Platform   *Platform     `xml:"Platform,omitempty"`
+	Location   *LocationType `xml:"Location,omitempty"`
+	Latitude   float64       `xml:"Latitude"`
+	Longitude  float64       `xml:"Longitude"`
+	ReportDate string        `xml:"ReportDate"`
+	Elevation  float64       `xml:"Elevation,omitempty"`
+	Altitude   float64       `xml:"Altitude,omitempty"`
+	Course     string        `xml:"Course"`
+	Speed      string        `xml:"Speed"`
+}
+
+// ToARESCoT translates a PositionReport to an ARES compatible format
+func (pr PositionReport) ToARESCoT() (*cot.Event, error) {
+	event := cot.Event{}
+	detail := cot.Detail{}
+	uid := cot.UID{}
+
+	event.How = "AIDE-DDS Translator"
```

Here's the defaults chosen for some of the more obscure CoT fields in Thesues (I remember getting these from somewhere). Were there other places in AIDE code where we set these to arbitrary strings like this?

  "defaultPointLE" : -1,
  "defaultPointCE" : -1,
  "defaultEventAccess" : "",
  "defaultEventHow" : "m-r",
  "defaultEventOpex" : "o",
  "defaultEventQos" : "1-r-c",